message/infrastructure/googlecloud/subscriber.go

```diff
@@ -0,0 +1,221 @@
+package googlecloud
+
+import (
+	"context"
+	"fmt"
+	"runtime"
+	"sync"
+
+	"google.golang.org/api/option"
+
+	"cloud.google.com/go/pubsub"
+	"github.com/ThreeDotsLabs/watermill"
+	"github.com/ThreeDotsLabs/watermill/message"
+	"github.com/pkg/errors"
+)
+
+var (
+	ErrSubscriberClosed = errors.New("subscriber is closed")
+)
+
+type subscriber struct {
+	closing chan struct{}
+	closed  bool
+
+	allSubscriptionsWaitGroup sync.WaitGroup
+	activeSubscriptions       map[string]struct{}
+	activeSubscriptionsLock   sync.RWMutex
+
+	client *pubsub.Client
+	config SubscriberConfig
+
+	unmarshaler Unmarshaler
+	logger      watermill.LoggerAdapter
+}
+
+type SubscriberConfig struct {
+	SubscriptionName            string
+	ProjectID                   string
+	CreateSubscriptionIfMissing bool
+
+	ConsumersCount int
```

Comment:
ConsumersCount is not used anywhere (and is probably not needed?)

Reply:
True, it's a leftover from the early prototype. The google client library takes care of spawning goroutines itself.

Reply:
a183611c6d4e3116d5cd766745f072da6d08f2f3